src/nimble.nim

```diff
@@ -84,8 +88,27 @@ proc processFreeDependencies(pkgInfo: PackageInfo, options: Options):
           [pkgInfo.basicInfo.name, $pkgInfo.basicInfo.version],
           priority = HighPriority)
 
-  var reverseDependencies: seq[PackageBasicInfo] = @[]
-  for dep in pkgInfo.requires:
+  var
+    reverseDependencies: seq[PackageBasicInfo] = @[]
+    requirements = pkgInfo.requires
+
+  template addTaskRequirements =
+    ## Adds all task requirements to list of requirements
+    for task, requires in pkgInfo.taskRequires:
+      requirements &= requires
+
+  # Check what task level dependencies need to be added
+  if isMain:
+    if task in pkgInfo.taskRequires:
+      # If this is the main file then add its needed requirements for running a task.
+      requirements &= pkgInfo.taskRequires[task]
+    elif options.action.typ in {actionLock, actionSync}:
```

Comment:
Why do these requirements (which are optional) have to be added to the lock file?

Reply:
They are only added to the lock file when generating/syncing it so that if someone wants to run the task then it will use the same locked version. When loading the lock file the task dependencies are skipped unless running the task (So they stay optional)

Reply:
Makes sense.